src/rely/TestResult.re

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */;
+open Common.Collections;
+
+type status =
+  | Passed
+  | Failed(string, option(Printexc.location), string)
+  | Exception(exn, option(Printexc.location), string);
+
+type time =
+  | Milliseconds(int);
+
+type testResult = {
+  path: TestPath.test,
+  duration: option(time),
+  testStatus: status,
+  title: string,
+  fullName: string,
+};
+
+type pendingTestResult = {
+  path: TestPath.test,
+  runTest: unit => unit,
+};
+
+type intermediateTestResult =
+  | PendingTestResult(pendingTestResult)
+  | FinalTestResult(testResult);
+
+type describeResult = {
+  testResults: list(testResult),
+  path: TestPath.describe,
+  describeResults: list(describeResult),
+  duration: option(time),
+};
+
+type snapshotSummary = {
+  numCreatedSnapshots: int,
+  numRemovedSnapshots: int,
+  numUpdatedSnapshots: int,
+};
+
+module TestSuiteResult = {
+  type t = {
+    numFailedTests: int,
```

Comment:
It's weird that this information is redundant with what's in `testResults`. I guess Jest does the same thing.